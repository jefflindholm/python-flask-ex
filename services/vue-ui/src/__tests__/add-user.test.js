import { mount } from '@vue/test-utils';
import { createRenderer } from 'vue-server-renderer'
import AddUser from '../components/add-user';

test('UsersList renders properly', () => {
  const vm = mount(AddUser);
  const inputs = vm.findAll('input');
  expect(inputs).toHaveLength(3);
  expect(inputs.at(0).attributes().name).toBe('username');
  expect(inputs.at(1).attributes().name).toBe('email');
  expect(inputs.at(2).attributes().type).toBe('submit');
});


test('UsersList renders correct snapshot', () => {
  const renderer = createRenderer();
  const vm = mount(AddUser);
  renderer.renderToString(vm.vm, (err, str) => {
    if (err) throw new Error(err);
    expect(str).toMatchSnapshot();
  });
});
