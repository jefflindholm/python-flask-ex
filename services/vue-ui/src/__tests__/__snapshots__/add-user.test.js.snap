// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders correct snapshot 1`] = `
<form>
  <div class="form-group">
    <input type="text" required="required" name="username" placeholder="Enter a username" value="" class="form-control input-lg">
    <input type="text" required="required" name="email" placeholder="Enter an email address" value="" class="form-control input-lg">
    <input type="submit" value="Submit" class="btn btn-primary btn-lg btn-block">
  </div>
</form>
`;
